### PR TITLE
Added http connect Autest with proxy verifier

### PR DESCRIPTION
**Pull request by lzx404243**
Added an Autest with proxy verifier to generate traffic that executes the http `CONNECT` request.

### CODE REVIEW

File: tests/gold_tests/connect/replays/connect.replay.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      - client-request:
+          method: GET
+          version: "1.1"
+          url: /get
+          headers:
+            fields:
+              - [uuid, 2]
+              - [Host, www.example.com]
+        server-response:
+          status: 200
```
Comment (bneradt) [inline tests/gold_tests/connect/replays/connect.replay.yaml:53]: Let's add some `X-Response` header for this proxied transaction and add a `proxy-response` verification that the client gets that back. The content will be opaque to ATS, but the proxy-verifier server should still function as normal and the client should get its response back.

File: tests/gold_tests/connect/connect.test.py [hunk 2]
```diff
@@ -106,3 +107,61 @@ def run(self):
...
+            description="Verify the server gets the second request.")
+
+    def setupTS(self):
+        self.ts = Test.MakeATSProcess("connect-ts")
+
+        self.ts.Disk.records_config.update({
+            'proxy.config.diags.debug.enabled': 1,
+            'proxy.config.diags.debug.tags': 'http',
+            'proxy.config.http.server_ports': f"{self.ts.Variables.port}",
+            'proxy.config.http.connect_ports': f"{self.server.Variables.http_port}",
```
Comment (bneradt) [inline tests/gold_tests/connect/connect.test.py:140]: I don't think these need to be strings. Can you try without making these string literals and see whether that works?

**Review by bneradt — CHANGES_REQUESTED** [review]
Looks good. Thank you for adding this.

File: tests/gold_tests/connect/connect.test.py (hunk 2 above)
Comment (lzx404243) [reply tests/gold_tests/connect/connect.test.py:140]: Tried that but that didn't work out. I think they take string type to support specifying a range of ports.

File: tests/gold_tests/connect/replays/connect.replay.yaml (hunk 1 above)
Comment (lzx404243) [reply tests/gold_tests/connect/replays/connect.replay.yaml:53]: Good point! Added those

File: tests/gold_tests/connect/connect.test.py (hunk 2 above)
Comment (bneradt) [reply tests/gold_tests/connect/connect.test.py:140]: Gotcha. Thanks for trying it out.

**Review by bneradt — APPROVED** [review]
Looks good.

**Review by randall — APPROVED** [review]

**Pull request merged**